zilliqa/src/block_store.rs

```diff
@@ -103,6 +124,13 @@ impl BlockStore {
             .peers
             .iter()
             .filter(|(_, peer)| peer.highest_known_view >= view)
+            // If the last request failed, don't send requests to this peer for 10 seconds.
+            .filter(|(_, peer)| {
+                peer.last_request_failed_at
+                    .and_then(|at| at.elapsed().ok())
+                    .map(|time_since| time_since > Duration::from_secs(10))
```

Comment:
Perhaps split this constant out into however you're managing constants these days? Having it embedded in the code makes it kinda hard to see.